app/jobs/ingest_file_job.rb

```diff
@@ -2,31 +2,39 @@ class IngestFileJob < ActiveJob::Base
   queue_as CurationConcerns.config.ingest_queue_name
 
   # @param [FileSet] file_set
-  # @param [String] filename
+  # @param [String] filename the cached file within the CurationConcerns.config.working_path
   # @param [String,NilClass] mime_type
   # @param [User] user
   # @param [String] relation ('original_file')
   def perform(file_set, filename, mime_type, user, relation = 'original_file')
-    file = File.open(filename, "rb")
+    local_file = File.open(filename, "rb")
     # If mime-type is known, wrap in an IO decorator
     # Otherwise allow Hydra::Works service to determine mime_type
     if mime_type
-      file = Hydra::Derivatives::IoDecorator.new(file)
-      file.mime_type = mime_type
-      file.original_name = File.basename(filename)
+      local_file = Hydra::Derivatives::IoDecorator.new(local_file)
+      local_file.mime_type = mime_type
+      local_file.original_name = File.basename(filename)
     end
 
     # Tell AddFileToFileSet service to skip versioning because versions will be minted by VersionCommitter (called by save_characterize_and_record_committer) when necessary
-    Hydra::Works::AddFileToFileSet.call(file_set, file, relation.to_sym, versioning: false)
+    Hydra::Works::AddFileToFileSet.call(file_set,
+                                        local_file,
+                                        relation.to_sym,
+                                        versioning: false)
 
     # Persist changes to the file_set
     file_set.save!
     file_set.in_works.each do |work|
       work.pending_uploads.where(file_name: Pathname.new(filename).basename.to_s).destroy_all
     end
 
+    repository_file = file_set.send(relation.to_sym)
+
     # Do post file ingest actions
-    CurationConcerns::VersioningService.create(file_set.send(relation.to_sym), user)
-    CharacterizeJob.perform_later(file_set, filename)
+    CurationConcerns::VersioningService.create(repository_file, user)
+
+    # TODO: this is a problem, the file may not be available at this path on another machine.
+    # It may be local, or it may be in s3
```

Comment:
OK, that's good to know — I hope we'll be able to update CC (or hydra_derivs) to pull files from Fedora if they're not on local disk.